src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/acl_editor.cljs

```diff
@@ -24,78 +24,114 @@
+                (:users-not-found @state))])
+    :ok-button {:text "Invite" :onClick #(react/call :persist-acl component true)}}])
+
 (react/defc AclEditor
   {:render
    (fn [{:keys [props state this]}]
      (if (or (:non-project-owner-acl-vec @state) (:project-owner-acl-vec @state))
-       (render-acl-content props state this)
+       (if (:offering-invites? @state)
+         (render-invite-offer props state this)
+         (render-acl-content props state this))
        [:div {:style {:padding "2em"}}
         (if (:load-error @state)
           (style/create-server-error-message (:load-error @state))
           [comps/Spinner {:text "Loading Permissions..."}])]))
    :persist-acl
-   (fn [{:keys [props state refs this]}]
+   (fn [{:keys [props state refs this]} invite-new?]
      (swap! state dissoc :save-error :validation-error)
      (let [filtered-acl (->> (concat (:project-owner-acl-vec @state) (:non-project-owner-acl-vec @state))
                              (map #(dissoc % :read-only?))
                              (map #(update-in % [:email] clojure.string/trim))
                              (filter #(not (empty? (:email %)))))
-           fails (apply input/validate refs (map #(str "acl-key" %) (range (count (:non-project-owner-acl-vec @state)))))]
+           fails (apply input/validate refs (filter
+                                             #(contains? @refs %)
+                                             (map #(str "acl-key" %) (range (count (:non-project-owner-acl-vec @state))))))]
        (if fails
          (swap! state assoc :validation-error fails)
          (do
            (swap! state assoc :saving? true)
            (endpoints/call-ajax-orch
-            {:endpoint (endpoints/update-workspace-acl (:workspace-id props))
+            {:endpoint (endpoints/update-workspace-acl (:workspace-id props) invite-new?)
              :headers utils/content-type=json
              :payload filtered-acl
              :on-done (fn [{:keys [success? get-parsed-response]}]
                         (swap! state dissoc :saving?)
-                        (if success?
-                          (do
-                            ((:request-refresh props))
-                            (modal/pop-modal))
-                          (swap! state assoc :save-error (get-parsed-response false))))})))))
+                        (if (seq (:usersNotFound (get-parsed-response)))
+                          (react/call :offer-user-invites this (get-parsed-response))
+                          (if success?
+                            (do
+                              ((:request-refresh props))
+                              (modal/pop-modal))
+                            (swap! state assoc :save-error (get-parsed-response false)))))})))))
+   :offer-user-invites
+   (fn [{:keys [state]} parsed-response]
+     (swap! state assoc :users-not-found (:usersNotFound parsed-response))
+     (swap! state assoc :offering-invites? true))
```

Comment:
Each `swap!` will trigger a re-render, so do both of these in one step to have just one rerender.